admin/server-timing.php

```diff
@@ -168,3 +183,56 @@ class="large-text code"
 	</p>
 	<?php
 }
+
+/**
+ * Renders a checkbox for enabling output buffering for Server-Timing.
+ *
+ * @since n.e.x.t
+ */
+function perflab_render_server_timing_page_output_buffer_checkbox() {
+	$slug           = 'output_buffering';
+	$field_id       = "server_timing_{$slug}";
+	$field_name     = PERFLAB_SERVER_TIMING_SETTING . '[' . $slug . ']';
+	$description_id = "{$field_id}_description";
+	$has_filter     = has_filter( 'perflab_server_timing_use_output_buffer' );
+	$is_enabled     = perflab_server_timing_use_output_buffer();
+
+	?>
+	<fieldset>
+		<legend class="screen-reader-text">
+			<?php esc_html_e( 'Output Buffering', 'performance-lab' ); ?>
+		</legend>
```

Comment:
See my feedback above. You can simplify how this field is rendered by using a dedicated section and heading. Additionally, a `fieldset` isn't really useful around a single checkbox, it makes more sense when it's a group of checkboxes that somehow work together.

Reply:
I would agree, but the PerfLab plugin uses the same `fieldset` approach on its settings screen:

![image](https://github.com/WordPress/performance/assets/134745/fe65a52b-62a4-4dd9-82cb-51b4c8108ec1)

```html
<fieldset>
		<legend class="screen-reader-text">
			Dominant Color Images		</legend>
		<label for="module_images/dominant-color-images_enabled">
							<input type="checkbox" id="module_images/dominant-color-images_enabled" name="perflab_modules_settings[images/dominant-color-images][enabled]" aria-describedby="module_images/dominant-color-images_description" value="1" checked="checked">
				Enable Dominant Color Images					</label>
		<p id="module_images/dominant-color-images_description" class="description">
			Adds support to store the dominant color of newly uploaded images and create a placeholder background of that color.		</p>
	</fieldset>
```


This is the same markup used  for other admin screens, like on General settings:

![image](https://github.com/WordPress/performance/assets/134745/c9fd763e-ba57-4a56-afae-1ce66141a931)

```html
<fieldset><legend class="screen-reader-text"><span>
	Membership</span></legend><label for="users_can_register">
<input name="users_can_register" type="checkbox" id="users_can_register" value="1">
	Anyone can register</label>
</fieldset>
```

And on Reading settings:

![image](https://github.com/WordPress/performance/assets/134745/aa904769-83ae-4846-a4be-1f98c303467b)

```html
<fieldset>
	<legend class="screen-reader-text"><span>
		Search engine visibility	</span></legend>
	<label for="blog_public"><input name="blog_public" type="checkbox" id="blog_public" value="0">
	Discourage search engines from indexing this site</label>
	<p class="description">It is up to search engines to honor this request.</p>
</fieldset>
```

That being said, there are other solitary checkboxes that lack the `fieldset`, including on Discussion settings:

![image](https://github.com/WordPress/performance/assets/134745/be36caf9-e055-4228-ab18-9d8e0b72b589)

```html
<td>
	<label for="show_avatars">
		<input type="checkbox" id="show_avatars" name="show_avatars" value="1" checked="checked">
		Show Avatars	</label>
</td>
```

Reply:
There is also a bit of styling present that only applies when the `fieldset` is present:

![image](https://github.com/WordPress/performance/assets/134745/a0b9d227-9dc6-423a-9696-0cd7f31bd0ac)

Without `fieldset`:

![image](https://github.com/WordPress/performance/assets/134745/cc26071b-a763-4329-9959-ce27bcc9f40d)

With `fieldset`:

![image](https://github.com/WordPress/performance/assets/134745/09b172bb-2917-45d4-9d5d-ea22ac229070)

Reply:
Fair point, there are clearly some inconsistencies. I was primarily referring to it because we don't need the fieldset approach anymore if the section has a heading anyway. The fieldset + legend approach doesn't help at all in this case because the "Output Buffering" text is already used as heading anyway. That would mean a screen reader would have "Output Buffering" twice, from the heading and then from the legend, which would be pointless.

I think we should remove the `legend` for that reason, and then using `fieldset` doesn't really provide a value either, other than the default styling tweak, which is very minor though, and arguably irrelevant if we don't _display_ the legend anyway. I would suggest to simply use a `div.output-buffering` if we need a wrapper element to target everything within.